### PR TITLE
[MTG-331]feat :additional tests

### PR DESCRIPTION
**Pull request by RequescoS**

### CODE REVIEW

**Review by StanChe — APPROVED** [review]
💪

File: programs/mpl-staking/tests/test_claim.rs [hunk 1]
```diff
@@ -259,517 +300,200 @@ async fn successeful_claim() -> Result<(), TransportError> {
...
+            &claim_setup.voter_authority_ata,
+            &claim_setup.context.rewards.program_id,
+            &claim_setup.registrar,
+            &claim_setup.mint_governance.address,
+            &claim_setup.proposal.address,
+            &claim_setup.voter,
             &vote_record,
         )
         .await
-        .assert_on_chain_err(MplStakingError::TokenflowRestricted);
```
Comment (kstepanovdev) [inline programs/mpl-staking/tests/test_claim.rs:500]: afaik claiming will return errors from the `Staking`. i guess `rewards` even doesn't have such errors.

File: programs/mpl-staking/tests/test_claim.rs [hunk 2]
```diff
@@ -259,517 +300,200 @@ async fn successeful_claim() -> Result<(), TransportError> {
...
+            &claim_setup.context.rewards.program_id,
+            &claim_setup.registrar,
+            &claim_setup.mint_governance.address,
+            &claim_setup.proposal.address,
+            &claim_setup.voter,
+            &vote_record,
         )
-        .await?;
+        .await
+        .expect_err("Claiming is restricted by Rewards program");
```
Comment (kstepanovdev) [inline programs/mpl-staking/tests/test_claim.rs:411]: Is it? Probably, the `staking` itself should return errors.

**Review by kstepanovdev — CHANGES_REQUESTED** [review]
Awesome changes. However, we might want to catch specific errors instead of capturing some errors that occurred in general if it's possible.

**Review by kstepanovdev — APPROVED** [review]
LGTM. Thank you for the refactoring.

**Pull request merged**